### ISSUE TITLE
Application is continuously requesting unauthorized ressource `external.svg` / Authentication missing for app / external icons

### ISSUE THREAD

**Issue by Somebodyisnobody** [nextcloud/desktop#2371]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
When Nextcloud desktop get's a 401 back while requesting external.svg the application should send authorization headers with the next request.

### Actual behaviour
Nextcloud desktop just sends a new unauthorized request.

### Steps to reproduce
1. Run nextcloud desktop
2. Filter your server logs for "/apps/external/icons/external.svg"

### Client configuration
Client version:
301

Operating system: Win10.2004 64Bit

OS language: EN

Qt version used by client package (Linux only, see also Settings dialog): Not applicable

Client package (From Nextcloud or distro) (Linux only): Not applicable

Installation path of client: Standard

### Server configuration

Nextcloud version: 18.0.7

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: -

Output of `nextcloud --logdebug --logwindow` or `nextcloud --logdebug --logfile log.txt`
(On Windows using `cmd.exe`, you might need to first `cd` into the Nextcloud directory)
(See also https://docs.nextcloud.com/desktop/2.3/troubleshooting.html#log-files)

> 2020-09-06 12:03:44:451 [ info nextcloud.sync.credentials.webflow ]:	request finished
2020-09-06 12:03:44:452 [ debug nextcloud.gui.account.state ]	[ OCC::AccountState::slotEtagResponseHeaderReceived ]:	New navigation apps ETag Response Header received  "W/\"23cdbd834a8b1f461cbc*********\""
2020-09-06 12:03:44:455 [ debug nextcloud.sync.networkjob ]	[ OCC::AbstractNetworkJob::slotFinished ]:	Network job OCC::OcsNavigationAppsJob finished for "ocs/v2.php/core/navigation/apps"
2020-09-06 12:03:44:492 [ warning default ]:	qrc:/qml/src/gui/tray/Window.qml:426:39: QML MenuItem: Binding loop detected for property "width"
2020-09-06 12:03:44:533 [ warning default ]:	qrc:/qml/src/gui/tray/Window.qml:426:39: QML MenuItem: Binding loop detected for property "width"
2020-09-06 12:03:44:534 [ warning default ]:	qrc:/qml/src/gui/tray/Window.qml:426:39: QML MenuItem: Binding loop detected for property "width"
2020-09-06 12:03:44:690 [ warning default ]:	file:///C:/Program Files (x86)/Nextcloud/QtQuick/Controls.2/MenuItem.qml: QML IconImage: Host requires authentication
2020-09-06 12:03:44:691 [ warning default ]:	file:///C:/Program Files (x86)/Nextcloud/QtQuick/Controls.2/MenuItem.qml: QML IconImage: Host requires authentication
2020-09-06 12:03:44:971 [ warning default ]:	file:///C:/Program Files (x86)/Nextcloud/QtQuick/Controls.2/MenuItem.qml: QML IconImage: Host requires authentication
2020-09-06 12:03:44:972 [ warning default ]:	file:///C:/Program Files (x86)/Nextcloud/QtQuick/Controls.2/MenuItem.qml: QML IconImage: Host requires authentication

2. Web server error log:
[messages_export.txt](https://github.com/nextcloud/desktop/files/5179384/messages_export.txt) (In line 31 I closed NC desktop to be sure that the application causes that error)
![Screenshot_2020-09-06](https://user-images.githubusercontent.com/35230554/92323100-3ff2d300-f036-11ea-8b58-711447fc249e.png)

3. Server logfile: nextcloud log (data/nextcloud.log): -

**Comment by Somebodyisnobody** [comment]
As a workarround I disabled now https://github.com/nextcloud/external/

**Comment by er-vin** [comment]
@nickvergessen hm... could it be due to something with the application itself? I don't quite get why we have such an error for that resource while it seems to work for other applications.

@Somebodyisnobody so in the application menu in the main dialog, you'd have no icon for the external application?

**Comment by Somebodyisnobody** [comment]
Nope there was no such icon in the new nextcloud-desktop ui.

**Comment by nickvergessen** [comment]
Yeah the endpoint is not accessible unauthorized. I can change that, but it would be nicer to not leak all those images to unauthenticated users. How do you decide whetther to send the auth header/cookies and when not?

**Comment by er-vin** [comment]
Well it looks like we don't for any of the application icons... but better fix that on the client end and send it every time maybe?

**Comment by Somebodyisnobody** [comment]
In my eyes it's a problem of nextcloud desktop: The application must be able to handle and react on 401 responses.

**Comment by nickvergessen** [comment]
> Well it looks like we don't for any of the application icons.

Well I guess the external app is the only app that is not a real image file on the disk, but a route that renders an image. But yeah sending it all the time shouldn't hurt.

**Comment by acsfer** [comment]
In our case, this affects every single icon on the External sites.

```
"GET /apps/external/icons/mail.svg" 401 
"GET /apps/external/icons/desktop.svg" 401
"GET /apps/external/icons/users.svg" 401
"GET /apps/external/icons/groups.svg" 401
...
```
This hurts the logs (+12K lines per day...) so for now, we've completely disable these log lines with a regex.

**Comment by hostingnuggets** [comment]
Please do not close issue.

**Comment by hostingnuggets** [comment]
Is this fixed?

**Comment by FlexW** [comment]
> Is this fixed?

No

**Comment by Rello** [comment]
Hello, 
please retry with a recent version and let us know in a new ticket if this is still valid.
thank you
